qtcodes/circuits/rotated_surface.py

```diff
@@ -302,13 +327,13 @@ def extract_final_stabilizer_and_logical_readout_z(
             z_stabilizer = str(stabilizer_val) + z_stabilizer
 
         stabilizer_str = (
-            previous_syndrome_string[: int(self.params["num_syn"])] + z_stabilizer
+            previous_syndrome_string[: self.params["num_syn"][self.SYNZ]] + z_stabilizer
```

Comment:
So, this should use `self.SYNX`, since we are taking the X syndrome from the previous syndrome string!

I've fixed it, but this shows that we should really have tests that involve `lattice_readout_z`, otherwise it's hard to catch bugs like this. 

Let's add these tests in a separate PR. Here's an issue to track this work #61

Reply:
Thanks. Good find!